src/algorithms/sorting/short_bubble_sort.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+var Comparator = require('../../util/comparator');
+
+/**
+ * short bubble sort algorithm
+ * worst: O(n^2) best: O(n)
+ */
+
+function shortBubbleSort(array, comparatorFn) {
+  var comparator = new Comparator(comparatorFn);
+  var length = array.length;
+  var i = 0;
+
+  for (i; i < length; i++) {
+    var current = array[i];
+    var next = array[i+1];
+
+    if (next === undefined) break;
+    if (comparator.lessThanOrEqual(current, next)) continue;
```

Comment:
instead of using continue, I'd suggest having a if `comparator.greaterThan` and the code below inside the block. It makes the flow easier to read IMHO and maybe add a comment explaining why decrement `i`